### PR TITLE
Word confusions and formatting

### PR DESCRIPTION
**Pull request by tshastry**
Adds better support for testing discovery with chat transcripts.

### CODE REVIEW

File: test_discovery.py [hunk 1]
```diff
@@ -34,6 +36,8 @@
 BAD_EXIT_CODE = 1
 GOOD_EXIT_CODE = 0
 
+UNFORMATTED_CHARS = set(list("abcdefghijklmnopqrstuvwxyz-' "))
```
Comment (mgoldey) [inline test_discovery.py]:
```suggestion
UNFORMATTED_CHARS = set("abcdefghijklmnopqrstuvwxyz-' ")
```

File: test_discovery.py [hunk 2]
```diff
@@ -121,7 +125,14 @@ def submit_transcript(transcript):
     """
     Submits a transcript to Discovery
     """
-    data = {"transcript": transcript}
+    data = {"word_confusions": [{w: "1.0"} for w in transcript.split(" ")]}
+    
+    if len(set(list(transcript)).symmetric_difference(UNFORMATTED_CHARS)) > 0:
```
Comment (mgoldey) [inline test_discovery.py]: Maybe 
```suggestion
    if any(map(lambda c: c not in UNFORMATTED_CHARS, transcript)):
```

File: test_discovery.py [hunk 3]
```diff
@@ -121,7 +125,14 @@ def submit_transcript(transcript):
     """
     Submits a transcript to Discovery
     """
-    data = {"transcript": transcript}
+    data = {"word_confusions": [{w: "1.0"} for w in transcript.split(" ")]}
+    
+    if len(set(list(transcript)).difference(UNFORMATTED_CHARS)) > 0:
```
Comment (tshastry) [inline test_discovery.py]:
```suggestion
    if any(map(lambda c: c not in UNFORMATTED_CHARS, transcript)):
```

**Comment by tshastry** [comment]
> Are we sure we want to force all discovery input to be unformatted?

This will only force it for intent classification, where classification performs better with unformatted text.

**Pull request merged**